lib/inventory_refresh/inventory_collection.rb

```diff
@@ -206,7 +209,19 @@ def store_updated_records(records)
     #
     # @param records [Array<ApplicationRecord, Hash>] list of stored records
     def store_deleted_records(records)
-      @deleted_records.concat(records_identities(records))
+      # TODO(lsmola) we need to move output streaming here, output_client of manageiq-messaging should be passed here
+      # and we should just keep track of a count here, not all ids. At least for deleting, the ids do not have upper
+      # limit, so the memory could bump a lot.
```

Comment:
We could yield the created/updated/deleted records in batches to the caller, just cutting off the deleted records at some fixed max doesn't seem right

Reply:
Or passing in a manageiq-messaging client, since then we're baking in how we're calling the object passed in in a way that won't be very applicable to other types of output clients

Reply:
Yeah I have the TODO there, we'll need to setup the streaming topic and push it there in batches right here, so the memory doesn't grow. I think it should be ok to have this as temporary solution?

That means passing the Streams objects to persister and IC.

Reply:
Lets drop it from here since it isn't related to mark&sweep

Reply:
ok